File: incomplete_fizz_buzz/java/test/FizzBuzzTest.java
import org.junit.Test;
import static org.junit.Assert.assertThat;
import static org.hamcrest.core.Is.is;

import java.util.List;
import java.util.ArrayList;

public class FizzBuzzTest {

  @Test
  public void printsMultipleOf3AsFizz() {
    PrinterSpy spy = new PrinterSpy();
    FizzBuzz.fizzBuzz(3, spy);
    List<String> results = new ArrayList<String>();
    results.add("Fizz");
    assertThat(spy.results(), is(results));
  }

  @Test
  public void printsMultipleOf5AsBuzz() {
    PrinterSpy spy = new PrinterSpy();
    FizzBuzz.fizzBuzz(5, spy);
    List<String> results = new ArrayList<String>();
    results.add("Buzz");
    assertThat(spy.results(), is(results));
  }

  @Test
  public void printsMultipleOf15AsFizzBuzz() {
    PrinterSpy spy = new PrinterSpy();
    FizzBuzz.fizzBuzz(15, spy);
    List<String> results = new ArrayList<String>();
    results.add("FizzBuzz");
    assertThat(spy.results(), is(results));
  }

  @Test
  public void printsNumberWhichIsNotMultipleOf3And5() {
    PrinterSpy spy = new PrinterSpy();
    FizzBuzz.fizzBuzz(1, spy);
    List<String> results = new ArrayList<String>();
    results.add("1");
    assertThat(spy.results(), is(results));
  }
}

File: incomplete_fizz_buzz/php/test/FizzBuzzTest.php
<?php

require_once('FizzBuzz.php');
require_once('PrinterSpy.php');

class FizzBuzzTest extends PHPUnit_Framework_TestCase
{

  public function test_prints_multiple_of_3_as_Fizz()
  {
    $spy = new PrinterSpy();
    FizzBuzz::fizz_buzz(3, $spy);
    $this->assertEquals(['Fizz'], $spy->results());
  }

  public function test_prints_multiple_of_5_as_Buzz()
  {
    $spy = new PrinterSpy();
    FizzBuzz::fizz_buzz(5, $spy);
    $this->assertEquals(['Buzz'], $spy->results());
  }

  public function test_prints_multiple_of_15_as_FizzBuzz()
  {
    $spy = new PrinterSpy();
    FizzBuzz::fizz_buzz(15, $spy);
    $this->assertEquals(['FizzBuzz'], $spy->results());
  }

  public function test_prints_number_which_is_not_multiple_of_3_and_5()
  {
    $spy = new PrinterSpy();
    FizzBuzz::fizz_buzz(1, $spy);
    $this->assertEquals(['1'], $spy->results());
  }
}
